go/vt/vtadmin/cluster/cluster.go

```diff
@@ -205,6 +206,42 @@ func (c *Cluster) parseTablet(rows *sql.Rows) (*vtadminpb.Tablet, error) {
 	return tablet, nil
 }
 
+// FindAllShardsInKeyspaceOptions modify the behavior of a cluster's
+// FindAllShardsInKeyspace method.
+type FindAllShardsInKeyspaceOptions struct {
+	// SkipDial indicates that the cluster can assume the vtctldclient has
+	// already dialed up a connection to a vtctld.
+	SkipDial bool
```

Comment:
I'm curious why `SkipDial` is exposed as an option given that all uses (so far) are `true`. Requiring the caller to dial a vtctld connection seems like a fair contract to me + it seems like it'd simplify the interface quite a bit. Let me know if I missed something though!

Reply:
I'm of two minds about requiring callers to know the details enough to call `Dial` (though, this option just does the same thing, only one thin layer removed).

I think long term what I want is for a cluster to be smart about the last time it dialed, and "just do the right thing", and the caller doesn't need to know or care. I'd like to leave this as-is for now and noodle on that some more.